File: shoring/desk/src/main/java/com/panopset/marin/oldswpkg/cards/pngmap/ChipPainter.java
package com.panopset.marin.oldswpkg.cards.pngmap;

import com.panopset.compat.AniGrid;
import com.panopset.fxapp.FontManagerFX;

public class ChipPainter {

    private final AniGrid aniGrid;

    public ChipPainter(int mapW, int mapH) {
       aniGrid = new AniGrid(mapW, mapH, 4, 2, FontManagerFX.getImgRatio());
    }

    public int getChipWidth() {
        return aniGrid.getDspWidth();
    }

    public int getChipHeight() {
        return aniGrid.getDspHeight();
    }

    public int[] getPaintDimensionsFX(Chip chip, int x, int y) {
        return aniGrid.getPaintDimensions(chip.getOffsetW(), chip.getOffsetH(), x, y);
    }

}

File: code/shoring/desk/src/main/kotlin/com/panopset/marin/oldswpkg/cards/pngmap/ChipPainter.kt
package com.panopset.marin.oldswpkg.cards.pngmap

import com.panopset.compat.AniGrid
import com.panopset.fxapp.FontManagerFX

class ChipPainter(mapW: Int, mapH: Int) {
    private val aniGrid: AniGrid

    init {
        aniGrid = AniGrid(mapW, mapH, 4, 2, FontManagerFX.imgRatio)
    }

    val chipWidth: Int
        get() = aniGrid.dspWidth
    val chipHeight: Int
        get() = aniGrid.dspHeight

    fun getPaintDimensionsFX(chip: Chip, x: Int, y: Int): IntArray {
        return aniGrid.getPaintDimensions(chip.offsetW, chip.offsetH, x, y)
    }
}
